lib/filters/http-headers.js

```diff
@@ -18,12 +18,22 @@ function httpHeaders (obj) {
   return obj
 }
 
+/**
+ * Filters authorization header and cookie _values_
+ *
+ * The filterSensitiveHeaders method filters individual
+ * cookie values
+ */
 function filterSensitiveHeaders (headers) {
   for (const key in headers) {
+    // undefined headers will be dropped when serialized as
+    // json, and don't need to be redacted.  If a cookie
+    // header is already redacted there's no need to parse its
+    // values.
+    if (headers[key] === undefined || REDACTED === headers[key]) {
```

Comment:
I was thinking about a `lib/constants.js` when I was writing that so, overkill or not I added it -- if only because once it's there it'll make future decisions about "should this be a constant" easier if we have a place for them.  For now I'd like to keep the two module separate with an eye towards unifying them when we do the 4.0 work.